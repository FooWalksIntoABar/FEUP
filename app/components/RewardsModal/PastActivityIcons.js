import React from 'react'
import { View, Image } from 'react-native'
import PropTypes from 'prop-types'
import * as Animatable from 'react-native-animatable'
import styles from '../../styles/RewardModal.style'
import Images from '../../assets/images/images'

export class PastActivityIcons extends React.Component {
  filterActivities = (_, index, array) => {
    if (this.props.currentActivity + 8 > array.length) return index >= array.length - 8
    else return index >= this.props.currentActivity && index < this.props.currentActivity + 8
  }

  render () {
    let greyedOutIcons = this.props.activities
      .filter(this.filterActivities)
      .map((activity, index) => activity.status
        ? (<View key={index} style={styles.grayedOutActivityIcon} />)
        : (<View key={index} style={styles.grayedOutActivityIcon}>
          <Image style={activity.photo ? styles.grayedOutActivityPhoto : styles.grayedOutActivityImage} recizeMode={'cover'} source={activity.photo ? Images[activity.photo] : Images[activity.image]} />
        </View>)
      )

    let pastActivityIcons = this.props.activities
      .filter(this.filterActivities)
      .reduce((accumulator, activity, index) => {
        if (activity.status) {
          return {
            counter: accumulator.counter + 1,
            views: [...accumulator.views, (<Animatable.View key={index} animation={'zoomIn'} delay={600 + accumulator.counter * 1000} style={[styles.pastActivityIcon, { backgroundColor: activity.color }]}>
              <Image style={activity.photo ? styles.pastActivityPhoto : styles.pastActivityImage} recizeMode={'cover'} source={activity.photo ? Images[activity.photo] : Images[activity.image]} />
            </Animatable.View>)]
          }
        } else {
          return {
            counter: accumulator.counter,
            views: [...accumulator.views, (<View key={index} style={styles.pastActivitySpacer} />)]
          }
        }
      }, { counter: 0, views: [] }).views

    return (
      <View style={styles.pastActivitiesContainer}>
        <View style={styles.greyedOutActivitiesContainer}>
          {greyedOutIcons}
        </View>
        <View style={styles.completedActivitiesContainer}>
          {pastActivityIcons}
        </View>
      </View>
    )
  }
}

PastActivityIcons.propTypes = {
  activities: PropTypes.array.isRequired,
  currentActivity: PropTypes.number.isRequired
}
