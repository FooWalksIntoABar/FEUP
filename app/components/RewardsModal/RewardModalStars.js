import React from 'react'
import { View } from 'react-native'
import PropTypes from 'prop-types'

import * as Animatable from 'react-native-animatable'

import styles from '../../styles/RewardModal.style'
import Images from '../../assets/images/images'

export class RewardModalStars extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      animations: [
        !(this.props.activities[this.props.currentActivity].status && this.props.activities[this.props.currentActivity].status.reward < 1),
        !(this.props.activities[this.props.currentActivity].status && this.props.activities[this.props.currentActivity].status.reward < 2),
        !(this.props.activities[this.props.currentActivity].status && this.props.activities[this.props.currentActivity].status.reward < 3)
      ]
    }

    this.stars = []
  }

  componentDidMount () {
    /* istanbul ignore next */
    setTimeout(() => {
      if (this.state.animations[0]) {
        this.stars[0].zoomIn(1000).then(() => {
          if (this.state.animations[1]) {
            this.stars[1].zoomIn(1000).then(() => {
              if (this.state.animations[2]) {
                this.stars[2].zoomIn(1000)
              }
            })
          }
        })
      }
    }, 400)

    for (let i = 0; i < 3; i++) {
      /* istanbul ignore next */
      setTimeout(() => { if (this.state.animations[i]) this.stars[i].bounceOutDown(2000) }, 4000 + i * 400)
      /* istanbul ignore next */
      setTimeout(() => { if (this.state.animations[i]) this.props.increaseProgress(i + 1) }, 4400 + i * 400)
    }
  }

  render () {
    return (
      <View style={styles.rewardContainer}>
        <Animatable.Image
          ref={ref => (this.stars.push(ref))}
          style={styles.icon}
          resizeMode={'center'}
          source={Images.ui.star} />
        <Animatable.Image
          ref={ref => (this.stars.push(ref))}
          style={[styles.icon, styles.iconCenter]}
          resizeMode={'center'}
          source={Images.ui.star} />
        <Animatable.Image
          ref={ref => (this.stars.push(ref))}
          style={styles.icon}
          resizeMode={'center'}
          source={Images.ui.star} />
      </View>
    )
  }
}

RewardModalStars.propTypes = {
  activities: PropTypes.array.isRequired,
  currentActivity: PropTypes.number.isRequired,
  increaseProgress: PropTypes.func.isRequired
}
