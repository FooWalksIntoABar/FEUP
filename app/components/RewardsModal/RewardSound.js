import React from 'react'
import PropTypes from 'prop-types'
import { Audio } from 'expo'

export class RewardSound extends React.Component {
  sound = null

  componentDidMount () {
    setTimeout(this.playSound, 400)
  }

  componentWillUnmount () {
    if (this.sound !== null) {
      this.sound.unloadAsync()
    }
  }

  playSound = async () => {
    this.sound = new Audio.Sound()
    try {
      if (this.props.success) await this.sound.loadAsync(require('../../assets/sounds/happy.wav'))
      else await this.sound.loadAsync(require('../../assets/sounds/sad.wav'))
      this.sound.playAsync()
      this.sound.setPositionAsync(0)
    } catch (error) { }
  }

  render () {
    return null
  }
}

RewardSound.propTypes = {
  success: PropTypes.bool.isRequired
}
