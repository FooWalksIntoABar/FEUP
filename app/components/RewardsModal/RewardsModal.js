import React from 'react'
import { View, Image, TouchableOpacity } from 'react-native'
import PropTypes from 'prop-types'

import Modal from 'react-native-modalbox'
import * as Animatable from 'react-native-animatable'
import Images from '../../assets/images/images'

import styles from '../../styles/RewardModal.style'
import { PastActivityIcons } from './PastActivityIcons'
import { RewardModalExperienceBar } from './RewardModalExperienceBar'
import { RewardModalStars } from './RewardModalStars'

export class RewardsModal extends React.Component {
  constructor (props) {
    super(props)

    this.state = {
      progress: (this.props.xp - this.props.level * 100) / 100
    }

    this.increaseProgress = this.increaseProgress.bind(this)
  }

  increaseProgress = (increase) => {
    this.setState({ progress: (this.props.xp - this.props.level * 100 + increase) / 100 })
  }

  returnURIorImage = (activity) => {
    // TODO: This checks whether the photo attribute is type URI and should probably just eventually be totally changed to URI.
    if (activity.photo !== undefined && activity.photo.includes('file://')) {
      return { uri: activity.photo }
    } else {
      return Images[activity.image]
    }
  }

  render () {
    let cardText = this.props.activities[this.props.currentActivity].status && this.props.activities[this.props.currentActivity].status.reward ? 'Atividade completada!' : 'Atividade falhada...'

    return (
      <Modal style={styles.rewardsModal} isOpen={this.props.activities[this.props.currentActivity].status !== undefined} backButtonClose={false} swipeToClose={false} backdropPressToClose={false}>
        <View style={styles.rewardContainer}>
          <Image style={[styles.icon, styles.iconGreyedOut]} resizeMode={'center'} source={Images.ui.star} />
          <Image style={[styles.icon, styles.iconGreyedOut, styles.iconCenter]} resizeMode={'center'} source={Images.ui.star} />
          <Image style={[styles.icon, styles.iconGreyedOut]} resizeMode={'center'} source={Images.ui.star} />
        </View>
        <RewardModalStars activities={this.props.activities} currentActivity={this.props.currentActivity} increaseProgress={this.increaseProgress} />
        <View style={styles.rewardCard}>
          <Animatable.Text
            animation={'fadeOut'}
            delay={2400}
            style={styles.cardTitle}>
            { cardText }
          </Animatable.Text>
          <RewardModalExperienceBar progress={this.state.progress} />
          <PastActivityIcons activities={this.props.activities} currentActivity={this.props.currentActivity} />
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity
            activeOpacity={0.8}
            style={[styles.button, styles.backButton]}
            onPress={this.props.backPress} >
            <Image
              style={styles.buttonImage}
              resizeMode={'contain'}
              source={Images.ui.cancel} />
          </TouchableOpacity>
          <TouchableOpacity
            activeOpacity={0.8}
            style={[styles.button, styles.nextButton]}
            onPress={this.props.nextPress} >
            <Image
              style={styles.buttonImage}
              resizeMode={'contain'}
              source={Images.ui.next} />
          </TouchableOpacity>
        </View>
      </Modal>
    )
  }
}

RewardsModal.propTypes = {
  activities: PropTypes.array.isRequired,
  currentActivity: PropTypes.number.isRequired,
  xp: PropTypes.number.isRequired,
  level: PropTypes.number.isRequired,
  nextPress: PropTypes.func.isRequired,
  backPress: PropTypes.func.isRequired
}
