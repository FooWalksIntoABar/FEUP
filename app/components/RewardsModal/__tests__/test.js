import 'react-native'
import React from 'react'
import { configure, shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'

import { PastActivityIcons } from '../PastActivityIcons'
import { RewardsModal } from '../RewardsModal'
import { RewardModalExperienceBar } from '../RewardModalExperienceBar'
import { RewardModalStars } from '../RewardModalStars'
import { activities, activitiesWithStatus } from '../../../__tests__/mockTestData'

configure({ adapter: new Adapter() })

describe('RewardsModal components', () => {
  it('renders PastActivityIcons correctly', () => {
    const wrapper = shallow(<PastActivityIcons
      currentActivity={0}
      activities={activities} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ activities: activitiesWithStatus })
    wrapper.unmount()
  })

  it('renders RewardModalExperienceBar correctly', () => {
    const wrapper = shallow(<RewardModalExperienceBar progress={0.5} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })

  it('renders RewardsModal correctly', () => {
    const wrapper = shallow(<RewardsModal
      currentActivity={0}
      activities={activities}
      xp={430}
      level={4}
      nextPress={jest.fn()}
      backPress={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.setProps({ activities: activitiesWithStatus })
    wrapper.instance().increaseProgress()
    wrapper.unmount()
  })

  it('renders RewardModalStars correctly', () => {
    jest.useFakeTimers()
    const wrapper = shallow(<RewardModalStars
      currentActivity={0}
      activities={activities}
      increaseProgress={jest.fn()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  })
})
