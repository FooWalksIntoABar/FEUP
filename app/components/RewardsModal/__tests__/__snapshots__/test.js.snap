// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RewardsModal components renders BonusModal correctly 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "transparent",
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
>
  <Component
    style={
      Object {
        "alignItems": "flex-end",
        "elevation": 7,
        "flexDirection": "row",
        "height": 313.5,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="center"
      source={1}
      style={
        Array [
          Object {
            "bottom": 103.5,
            "height": 150,
            "marginHorizontal": 5,
            "opacity": 0,
            "width": 150,
          },
          Object {
            "opacity": 1,
            "tintColor": "#ccc5b9",
          },
        ]
      }
    />
    <Image
      resizeMode="center"
      source={1}
      style={
        Array [
          Object {
            "bottom": 103.5,
            "height": 150,
            "marginHorizontal": 5,
            "opacity": 0,
            "width": 150,
          },
          Object {
            "opacity": 1,
            "tintColor": "#ccc5b9",
          },
          Object {
            "height": 180,
            "width": 180,
          },
        ]
      }
    />
    <Image
      resizeMode="center"
      source={1}
      style={
        Array [
          Object {
            "bottom": 103.5,
            "height": 150,
            "marginHorizontal": 5,
            "opacity": 0,
            "width": 150,
          },
          Object {
            "opacity": 1,
            "tintColor": "#ccc5b9",
          },
        ]
      }
    />
  </Component>
  <RewardModalStars
    activities={
      Array [
        Object {
          "status": 3,
        },
      ]
    }
    currentActivity={0}
    increaseProgress={[Function]}
  />
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3db3e2",
        "borderRadius": 30,
        "elevation": 6,
        "flexGrow": 1,
        "marginHorizontal": 180,
        "marginVertical": 120,
        "position": "relative",
      }
    }
  >
    <withAnimatable(Text)
      animation="fadeOut"
      delay={2400}
      direction="normal"
      iterationCount={1}
      iterationDelay={0}
      onAnimationBegin={[Function]}
      onAnimationEnd={[Function]}
      onTransitionBegin={[Function]}
      onTransitionEnd={[Function]}
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 60,
          "marginHorizontal": 30,
          "position": "absolute",
          "textAlign": "center",
          "textShadowColor": "rgba(0, 0, 0, 0.2)",
          "textShadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "textShadowRadius": 3,
          "top": 156,
        }
      }
      useNativeDriver={false}
    >
      Rotina Completa
    </withAnimatable(Text)>
    <RewardModalExperienceBar
      progress={-1}
    />
    <Component
      style={
        Object {
          "alignItems": "center",
          "bottom": 0,
          "elevation": 7,
          "flexDirection": "row",
          "justifyContent": "center",
          "padding": 60,
          "position": "absolute",
          "right": 0,
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.8}
        onPress={[MockFunction]}
        style={
          Array [
            Object {
              "alignItems": "center",
              "aspectRatio": 1,
              "borderRadius": 75,
              "elevation": 7,
              "flexDirection": "row",
              "height": 150,
              "justifyContent": "center",
            },
            Object {
              "backgroundColor": "#a4e76f",
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={1}
          style={
            Object {
              "height": "50%",
              "tintColor": "#fff",
              "width": "50%",
            }
          }
        />
      </TouchableOpacity>
    </Component>
    <RewardSound
      playSounds={true}
      success={true}
    />
  </Component>
</Component>
`;

exports[`RewardsModal components renders PastActivityIcons correctly 1`] = `
<Component
  style={
    Object {
      "bottom": 120,
      "position": "absolute",
    }
  }
>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 60,
      }
    }
  >
    <Component
      key="0"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ccc5b9",
          "borderRadius": 37.5,
          "elevation": 2,
          "height": 75,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "overflow": "hidden",
          "width": 75,
        }
      }
    />
    <Component
      key="1"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ccc5b9",
          "borderRadius": 37.5,
          "elevation": 2,
          "height": 75,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "overflow": "hidden",
          "width": 75,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "60%",
            "opacity": 0.3,
            "width": "60%",
          }
        }
      />
    </Component>
    <Component
      key="2"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ccc5b9",
          "borderRadius": 37.5,
          "elevation": 2,
          "height": 75,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "overflow": "hidden",
          "width": 75,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "60%",
            "opacity": 0.3,
            "width": "60%",
          }
        }
      />
    </Component>
    <Component
      key="3"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ccc5b9",
          "borderRadius": 37.5,
          "elevation": 2,
          "height": 75,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "overflow": "hidden",
          "width": 75,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "60%",
            "opacity": 0.3,
            "width": "60%",
          }
        }
      />
    </Component>
    <Component
      key="4"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ccc5b9",
          "borderRadius": 37.5,
          "elevation": 2,
          "height": 75,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "overflow": "hidden",
          "width": 75,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "60%",
            "opacity": 0.3,
            "width": "60%",
          }
        }
      />
    </Component>
    <Component
      key="5"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ccc5b9",
          "borderRadius": 37.5,
          "elevation": 2,
          "height": 75,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "overflow": "hidden",
          "width": 75,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "60%",
            "opacity": 0.3,
            "width": "60%",
          }
        }
      />
    </Component>
    <Component
      key="6"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ccc5b9",
          "borderRadius": 37.5,
          "elevation": 2,
          "height": 75,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "overflow": "hidden",
          "width": 75,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "60%",
            "opacity": 0.3,
            "width": "60%",
          }
        }
      />
    </Component>
    <Component
      key="7"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ccc5b9",
          "borderRadius": 37.5,
          "elevation": 2,
          "height": 75,
          "justifyContent": "center",
          "marginHorizontal": 5,
          "overflow": "hidden",
          "width": 75,
        }
      }
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": undefined,
          }
        }
        style={
          Object {
            "height": "60%",
            "opacity": 0.3,
            "width": "60%",
          }
        }
      />
    </Component>
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 60,
        "position": "absolute",
      }
    }
  >
    <withAnimatable(Component)
      animation="zoomIn"
      delay={600}
      direction="normal"
      iterationCount={1}
      iterationDelay={0}
      key="0"
      onAnimationBegin={[Function]}
      onAnimationEnd={[Function]}
      onTransitionBegin={[Function]}
      onTransitionEnd={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 37.5,
            "elevation": 2,
            "height": 75,
            "justifyContent": "center",
            "marginHorizontal": 5,
            "opacity": 1,
            "overflow": "hidden",
            "width": 75,
          },
          Object {
            "backgroundColor": "#7d84b2",
          },
        ]
      }
      useNativeDriver={false}
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "uri": "bedroom",
          }
        }
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      />
    </withAnimatable(Component)>
    <Component
      key="1"
      style={
        Object {
          "height": 75,
          "marginHorizontal": 5,
          "width": 75,
        }
      }
    />
    <Component
      key="2"
      style={
        Object {
          "height": 75,
          "marginHorizontal": 5,
          "width": 75,
        }
      }
    />
    <Component
      key="3"
      style={
        Object {
          "height": 75,
          "marginHorizontal": 5,
          "width": 75,
        }
      }
    />
    <Component
      key="4"
      style={
        Object {
          "height": 75,
          "marginHorizontal": 5,
          "width": 75,
        }
      }
    />
    <Component
      key="5"
      style={
        Object {
          "height": 75,
          "marginHorizontal": 5,
          "width": 75,
        }
      }
    />
    <Component
      key="6"
      style={
        Object {
          "height": 75,
          "marginHorizontal": 5,
          "width": 75,
        }
      }
    />
    <Component
      key="7"
      style={
        Object {
          "height": 75,
          "marginHorizontal": 5,
          "width": 75,
        }
      }
    />
  </Component>
</Component>
`;

exports[`RewardsModal components renders RewardModalExperienceBar correctly 1`] = `
<withAnimatable(Component)
  animation="fadeIn"
  delay={3400}
  direction="normal"
  iterationCount={1}
  iterationDelay={0}
  onAnimationBegin={[Function]}
  onAnimationEnd={[Function]}
  onTransitionBegin={[Function]}
  onTransitionEnd={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "gray",
        "borderRadius": 37.5,
        "elevation": 6,
        "height": 75,
        "overflow": "hidden",
        "position": "absolute",
        "top": 150,
        "width": 680,
      },
    ]
  }
  useNativeDriver={false}
>
  <ProgressBar
    animated={true}
    animationConfig={
      Object {
        "bounciness": 0,
      }
    }
    animationType="spring"
    borderRadius={0}
    borderWidth={0}
    color="#a4e76f"
    height={75}
    indeterminate={false}
    progress={0.5}
    useNativeDriver={false}
    width={null}
  />
  <Component
    style={
      Array [
        Object {
          "left": "25%",
        },
        Object {
          "borderRightColor": "#000",
          "borderRightWidth": 2,
          "height": 75,
          "opacity": 0.5,
          "position": "absolute",
        },
      ]
    }
  />
  <Component
    style={
      Array [
        Object {
          "left": "50%",
        },
        Object {
          "borderRightColor": "#000",
          "borderRightWidth": 2,
          "height": 75,
          "opacity": 0.5,
          "position": "absolute",
        },
      ]
    }
  />
  <Component
    style={
      Array [
        Object {
          "left": "75%",
        },
        Object {
          "borderRightColor": "#000",
          "borderRightWidth": 2,
          "height": 75,
          "opacity": 0.5,
          "position": "absolute",
        },
      ]
    }
  />
</withAnimatable(Component)>
`;

exports[`RewardsModal components renders RewardModalStars correctly 1`] = `
<Component
  style={
    Object {
      "alignItems": "flex-end",
      "elevation": 7,
      "flexDirection": "row",
      "height": 313.5,
      "justifyContent": "center",
      "overflow": "hidden",
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
>
  <withAnimatable(Image)
    delay={0}
    direction="normal"
    iterationCount={1}
    iterationDelay={0}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    onTransitionBegin={[Function]}
    onTransitionEnd={[Function]}
    resizeMode="center"
    source={1}
    style={
      Object {
        "bottom": 103.5,
        "height": 150,
        "marginHorizontal": 5,
        "opacity": 0,
        "width": 150,
      }
    }
    useNativeDriver={false}
  />
  <withAnimatable(Image)
    delay={0}
    direction="normal"
    iterationCount={1}
    iterationDelay={0}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    onTransitionBegin={[Function]}
    onTransitionEnd={[Function]}
    resizeMode="center"
    source={1}
    style={
      Array [
        Object {
          "bottom": 103.5,
          "height": 150,
          "marginHorizontal": 5,
          "opacity": 0,
          "width": 150,
        },
        Object {
          "height": 180,
          "width": 180,
        },
      ]
    }
    useNativeDriver={false}
  />
  <withAnimatable(Image)
    delay={0}
    direction="normal"
    iterationCount={1}
    iterationDelay={0}
    onAnimationBegin={[Function]}
    onAnimationEnd={[Function]}
    onTransitionBegin={[Function]}
    onTransitionEnd={[Function]}
    resizeMode="center"
    source={1}
    style={
      Object {
        "bottom": 103.5,
        "height": 150,
        "marginHorizontal": 5,
        "opacity": 0,
        "width": 150,
      }
    }
    useNativeDriver={false}
  />
</Component>
`;

exports[`RewardsModal components renders RewardSound correctly 1 1`] = `""`;

exports[`RewardsModal components renders RewardSound correctly 2 1`] = `""`;

exports[`RewardsModal components renders RewardsModal correctly 1`] = `
<ModalBox
  animationDuration={400}
  backButtonClose={false}
  backdrop={true}
  backdropColor="black"
  backdropContent={null}
  backdropOpacity={0.5}
  backdropPressToClose={false}
  coverScreen={false}
  easing={[Function]}
  isOpen={true}
  keyboardTopOffset={22}
  position="center"
  startOpen={false}
  style={
    Object {
      "backgroundColor": "transparent",
      "height": "100%",
      "position": "relative",
      "width": "100%",
    }
  }
  swipeThreshold={50}
  swipeToClose={false}
>
  <Component
    style={
      Object {
        "alignItems": "flex-end",
        "elevation": 7,
        "flexDirection": "row",
        "height": 313.5,
        "justifyContent": "center",
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "width": "100%",
      }
    }
  >
    <Image
      resizeMode="center"
      source={1}
      style={
        Array [
          Object {
            "bottom": 103.5,
            "height": 150,
            "marginHorizontal": 5,
            "opacity": 0,
            "width": 150,
          },
          Object {
            "opacity": 1,
            "tintColor": "#ccc5b9",
          },
        ]
      }
    />
    <Image
      resizeMode="center"
      source={1}
      style={
        Array [
          Object {
            "bottom": 103.5,
            "height": 150,
            "marginHorizontal": 5,
            "opacity": 0,
            "width": 150,
          },
          Object {
            "opacity": 1,
            "tintColor": "#ccc5b9",
          },
          Object {
            "height": 180,
            "width": 180,
          },
        ]
      }
    />
    <Image
      resizeMode="center"
      source={1}
      style={
        Array [
          Object {
            "bottom": 103.5,
            "height": 150,
            "marginHorizontal": 5,
            "opacity": 0,
            "width": 150,
          },
          Object {
            "opacity": 1,
            "tintColor": "#ccc5b9",
          },
        ]
      }
    />
  </Component>
  <RewardModalStars
    activities={
      Array [
        Object {
          "color": "#7d84b2",
          "image": "bed",
          "photo": "bedroom",
          "status": Object {
            "reward": 1,
          },
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 0,
          },
          "title": "Fazer a cama",
        },
        Object {
          "color": "#0e79b2",
          "image": "toothbrush",
          "time": Object {
            "goal": 10,
            "max": 15,
            "min": 1,
          },
          "title": "Lavar os dentes",
        },
        Object {
          "color": "#7fb800",
          "image": "socks",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Vestir",
        },
        Object {
          "color": "#37c1f0",
          "image": "shower",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar banho",
        },
        Object {
          "color": "#e43f6f",
          "image": "bag",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Preparar a mochila",
        },
        Object {
          "color": "#4bb3fd",
          "image": "sneakers",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Calçar os sapatos",
        },
        Object {
          "color": "#ff7f11",
          "image": "breakfast",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Tomar o pequeno almoço",
        },
        Object {
          "color": "#b0db43",
          "image": "comb",
          "time": Object {
            "goal": 60,
            "max": 120,
            "min": 10,
          },
          "title": "Pentear cabelo",
        },
      ]
    }
    currentActivity={0}
    increaseProgress={[Function]}
  />
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3db3e2",
        "borderRadius": 30,
        "elevation": 6,
        "flexGrow": 1,
        "marginHorizontal": 180,
        "marginVertical": 120,
        "position": "relative",
      }
    }
  >
    <withAnimatable(Text)
      animation="fadeOut"
      delay={2400}
      direction="normal"
      iterationCount={1}
      iterationDelay={0}
      onAnimationBegin={[Function]}
      onAnimationEnd={[Function]}
      onTransitionBegin={[Function]}
      onTransitionEnd={[Function]}
      style={
        Object {
          "color": "#fff",
          "fontFamily": "LinotteBold",
          "fontSize": 60,
          "marginHorizontal": 30,
          "position": "absolute",
          "textAlign": "center",
          "textShadowColor": "rgba(0, 0, 0, 0.2)",
          "textShadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "textShadowRadius": 3,
          "top": 156,
        }
      }
      useNativeDriver={false}
    >
      Atividade completada!
    </withAnimatable(Text)>
    <RewardModalExperienceBar
      progress={0.3}
    />
    <PastActivityIcons
      activities={
        Array [
          Object {
            "color": "#7d84b2",
            "image": "bed",
            "photo": "bedroom",
            "status": Object {
              "reward": 1,
            },
            "time": Object {
              "goal": 60,
              "max": 120,
              "min": 0,
            },
            "title": "Fazer a cama",
          },
          Object {
            "color": "#0e79b2",
            "image": "toothbrush",
            "time": Object {
              "goal": 10,
              "max": 15,
              "min": 1,
            },
            "title": "Lavar os dentes",
          },
          Object {
            "color": "#7fb800",
            "image": "socks",
            "time": Object {
              "goal": 60,
              "max": 120,
              "min": 10,
            },
            "title": "Vestir",
          },
          Object {
            "color": "#37c1f0",
            "image": "shower",
            "time": Object {
              "goal": 60,
              "max": 120,
              "min": 10,
            },
            "title": "Tomar banho",
          },
          Object {
            "color": "#e43f6f",
            "image": "bag",
            "time": Object {
              "goal": 60,
              "max": 120,
              "min": 10,
            },
            "title": "Preparar a mochila",
          },
          Object {
            "color": "#4bb3fd",
            "image": "sneakers",
            "time": Object {
              "goal": 60,
              "max": 120,
              "min": 10,
            },
            "title": "Calçar os sapatos",
          },
          Object {
            "color": "#ff7f11",
            "image": "breakfast",
            "time": Object {
              "goal": 60,
              "max": 120,
              "min": 10,
            },
            "title": "Tomar o pequeno almoço",
          },
          Object {
            "color": "#b0db43",
            "image": "comb",
            "time": Object {
              "goal": 60,
              "max": 120,
              "min": 10,
            },
            "title": "Pentear cabelo",
          },
        ]
      }
      currentActivity={0}
    />
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "bottom": 45,
        "elevation": 7,
        "flexDirection": "row",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={[MockFunction]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "borderRadius": 75,
            "elevation": 7,
            "flexDirection": "row",
            "height": 150,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#FFA23D",
            "marginRight": 180,
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": "50%",
            "tintColor": "#fff",
            "width": "50%",
          }
        }
      />
    </TouchableOpacity>
    <TouchableOpacity
      activeOpacity={0.8}
      onPress={[MockFunction]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "aspectRatio": 1,
            "borderRadius": 75,
            "elevation": 7,
            "flexDirection": "row",
            "height": 150,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#a4e76f",
          },
        ]
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          Object {
            "height": "50%",
            "tintColor": "#fff",
            "width": "50%",
          }
        }
      />
    </TouchableOpacity>
  </Component>
  <RewardSound
    playSounds={false}
    success={true}
  />
</ModalBox>
`;
